src/udx.c

```diff
@@ -687,6 +702,46 @@ process_data_packet (udx_stream_t *stream, int type, uint32_t seq, char *data, s
   udx__cirbuf_set(&(stream->incoming), (udx_cirbuf_val_t *) pkt);
 }
 
+static int
+relay_packet (udx_stream_t *stream, char *buf, ssize_t buf_len, int type, uint32_t seq, uint32_t ack) {
+  stream->seq = max_uint32(stream->seq, seq);
+
+  udx_stream_t *relay = stream->relay_to;
+
+  if (type & UDX_HEADER_DESTROY) {
+    stream->status |= UDX_STREAM_DESTROYED_REMOTE;
+    close_maybe(stream, UV_ECONNRESET);
```

Comment:
this is tricky actually, as this tears down the memory, so you are not allowed to use the stream afterwards if close fires (which it will here).

a solution could be to move this below the "try instant relay" and just don't queue the destroy packet if that fails

... we need timewait state

Reply:
`stream` isn't used after this block though, only `relay`. Am I missing something?

Reply:
ah, simplifies, but how do we know the relay wasn't destroyed as well in the callback from close? ie reentry

Reply:
That's a great point! I'll refactor so it's always called last.